stacker/tests/actions/test_build.py

```diff
@@ -261,6 +200,161 @@ def test_Raises_StackDoesNotExist_from_lookup_non_included_stack(self):
             build_action._generate_plan()
 
 
+class TestLaunchStack(TestBuildAction):
+    def setUp(self):
+        self.context = self._get_context()
+        self.provider = Provider(None, interactive=False, recreate_failed=True)
+        self.build_action = build.Action(self.context, provider=self.provider)
+
+        self.stack = mock.MagicMock()
+        self.stack.name = 'vpc'
+        self.stack.fqn = 'vpc'
+        self.stack.locked = False
+        self.stack_status = None
+
+        plan = self.build_action._generate_plan()
+        _, self.step = plan.list_pending()[0]
+        self.step.stack = self.stack
+
+        def patch_object(*args, **kwargs):
+            m = mock.patch.object(*args, **kwargs)
+            self.addCleanup(m.stop)
+            m.start()
+
+        def get_stack(name, *args, **kwargs):
+            if name != self.stack.name or not self.stack_status:
+                raise StackDoesNotExist(name)
+
+            return {'StackName': self.stack.name,
+                    'StackStatus': self.stack_status,
+                    'Tags': []}
+
+        patch_object(self.provider, 'get_stack', side_effect=get_stack)
+        patch_object(self.provider, 'update_stack')
+        patch_object(self.provider, 'create_stack')
+        patch_object(self.provider, 'destroy_stack')
+
+        patch_object(self.build_action, "s3_stack_push")
+
+    def _advance(self, new_provider_status, expected_status, expected_reason):
```

Comment:
holy crap I love this new test suite - thanks so much @danielkza, this is great.

One thing I would suggest - should we have tests for when `recreate_failed` is false?

Reply:
I think I might just set it to `False` instead, and just enabled it on the tests that actually care about it.

Reply:
Changed the default to `False`, and only enable it in the re-creation test.